R/effectsizecomputation.R

```diff
@@ -0,0 +1,900 @@
+        replace     = i == 1,
+        add.measure = TRUE,
+        data        = if (!is.null(dataOutput)) list(dataOutput)
+      )
+
+      newDataOutput <- try(do.call(metafor::escalc, escalcInput))
+    }
+
+    if (inherits(newDataOutput, "try-error")) {
+      errors[[paste0("i",i)]] <- list(
+        step  = i,
+        error = .escCleanErrorMessage(attr(newDataOutput, "condition")$message, effectSizeType)
+      )
+    } else {
+
+      # keep track of computation steps
+      # (needs to be done manually as the same effect size can be specified multiple times...)
+      newDataOutput$step <- NA
+      newDataOutput[["step"]][!is.na(newDataOutput[["yi"]])] <- i
+
+      if (is.null(dataOutput)) {
+        dataOutput <- newDataOutput
+      } else {
+        dataOutput[is.na(dataOutput[["yi"]]),] <- newDataOutput[is.na(dataOutput[["yi"]]),]
+      }
+    }
+  }
+
+  # create an empty list if nothing was computed
+  if (is.null(dataOutput))
+    dataOutput <- list()
+
+  attr(dataOutput, "errors") <- errors
+  return(dataOutput)
+}
+.escComputeSummaryTable <- function(jaspResults, dataset, options, dataOutput) {
+
+  # create summary table
+  computeSummary <- createJaspTable(title = gettext("Summary"))
+  computeSummary$dependOn(c("effectSizeType", "variables"))
+  computeSummary$position <- 1
+
+  jaspResults[["computeSummary"]] <- computeSummary
+
+  computeSummary$addColumnInfo(name = "step",          title = gettext("Step"),           type = "integer")
+  computeSummary$addColumnInfo(name = "effectSize",    title = gettext("Effect Size"),    type = "string")
+  computeSummary$addColumnInfo(name = "computed",      title = gettext("Computed"),       type = "integer")
+  computeSummary$addColumnInfo(name = "totalComputed", title = gettext("Total Computed"), type = "integer")
+
+  # compute summary
+  if (length(seq_along(options[["effectSizeType"]])) > 0) {
+
+    computeSummaryData <- lapply(seq_along(options[["effectSizeType"]]), function(i) {
+      list("step" = i, "effectSize" = options[["effectSizeType"]][[i]][["effectSize"]], "computed" = sum(dataOutput[["step"]] == i, na.rm = TRUE))
+    })
+    computeSummaryData <- do.call(rbind.data.frame, computeSummaryData)
+    computeSummaryData$totalComputed <- cumsum(computeSummaryData$computed)
+
+    # set the data
+    computeSummary$setData(computeSummaryData)
+
+    if (nrow(dataset) == sum(computeSummaryData[["computed"]]))
+      computeSummary$addFootnote(gettext("Effect sizes were successfully computed for each data entry."))
+    else
+      computeSummary$addFootnote(gettextf(
+        "Effect sizes were successfully computed for %1$i out of %2$i data entries.",
+        sum(computeSummaryData[["computed"]]),
+        nrow(dataset)))
+  }
+
+  computeErrors <- attr(dataOutput, "errors")
+  for (i in seq_along(computeErrors)) {
+    computeSummary$addFootnote(computeErrors[[i]]$error, symbol = gettextf("Error in Step %1$i:", computeErrors[[i]]$step))
+  }
+
```

Comment:
```suggestion
       if (nrow(dataset) == sum(computeSummaryData[["computed"]]))
      footnote <- gettext("Effect sizes were successfully computed for each data entry.")
    else
      footnote <- (gettextf(
        "Effect sizes were successfully computed for %1$i out of %2$i data entries.",
        sum(computeSummaryData[["computed"]]),
        nrow(dataset)))
  }

  computeErrors <- attr(dataOutput, "errors")
  for (i in seq_along(computeErrors)) {
    footnote <- gettextf("%1$s <b>Error</b> in step %2$i: %3$s",
                         footnote,
                         computeErrors[[i]]$step,
                         computeErrors[[i]]$error)
  }

  computeSummary$addFootnote(footnote)
```
How about this?

Reply:
Why? This would overwrite the overall information footnote and the errors in the individual steps. I don't really want that.

I think the best behavior is to say whether or not all effect sizes were computed (the first footnote) and then inform about a problem in each separate step if there was any.

Reply:
I dont know what you mean with overwrite. but this is what I see. I think it looks more clean. anyways, could you can keep as is
<img width="1392" alt="Screenshot 2024-11-03 at 19 42 22" src="https://github.com/user-attachments/assets/a23aa6c9-67d2-490e-8856-522ec62c4901">